### ISSUE TITLE
StopIteration

### ISSUE THREAD

**Issue by gydlcc** [Bartzi/stn-ocr#21]
[stn-ocr](https://user-images.githubusercontent.com/26570842/37317624-f0738fec-26a0-11e8-9752-2a371790fd2d.png)
![stn-ocr](https://user-images.githubusercontent.com/26570842/37319594-c312c428-26aa-11e8-8126-8ca65b0e8e74.png)

 Hello dear B:
   it is seems like only run 10 epochs
      how could i resolve this ?!

**Comment by Bartzi** [comment]
Yes running for only 10 epochs is intended behaviour. You can set the number of epochs with the command-line switch `--num-epochs`.

**Comment by gydlcc** [comment]
yeah, thank you, as in the picture i uploaded, i wonder that the information between Traceback and StopIteration, is that t means i can't plot the logs? i wanna plot logs, how could i solve it.

**Comment by Bartzi** [comment]
I think it can not plot the logs, because the log output of MXNet is not as expected (this changes from time to time, especially if you are using a version of MXNet, other than `0.9.3`).
In order to fix this you would have to check the log parsing code in `utils/plot_log.py` and determine whats different with the `log_output` and fix the code :sweat_smile:

**Comment by gydlcc** [comment]
alright, thank you pretty much

**Comment by gydlcc** [comment]
Hello there, sorry to bother you again. I noticed that your groundtruth file only include label and didn't include the groundtruth bbox coordinates, if my own data have groundtruth bbox coordinates, is that code still work?

**Comment by Bartzi** [comment]
I think it should still work, but you will need to separate the bbox coordinates from the labels.

**Comment by gydlcc** [comment]
Alright, thank you pretty much

**Comment by avasisht-celadon** [comment]
hi, I still get the above error with 100 epochs.
Could you please tell how I would solve it?
![capture](https://user-images.githubusercontent.com/43161667/48533185-72fc2d80-e8c9-11e8-97ef-7a56898241db.jpg)

**Comment by Bartzi** [comment]
It seems that the output of your training looks different than what is expected by the `log_plotter` are you sure that you are using MXNet in version `0.9.3`?

**Comment by avasisht-celadon** [comment]
Yes, it was not v0.9.3. 
I cloned mxnet again and checked out v0.9.3. Then, I issued "make" to build mxnet. But, I do get the error:-

Makefile:27: mshadow/make/mshadow.mk: No such file or directory
Makefile:28: /home/mxnet/dmlc-core/make/dmlc.mk: No such file or directory
Makefile:126: /home/mxnet/ps-lite/make/ps.mk: No such file or directory
make: *** No rule to make target '/home/mxnet/ps-lite/make/ps.mk'.  Stop.

I then did git clone --recursive ,  the same above error persists.

**Comment by avasisht-celadon** [comment]
I solved the above issue, by cloning the submodules. But, Now, ended up with lot of compilation errors.
These errors are absent in latest mxnet versions though.

**Comment by Bartzi** [comment]
Well, that is a problem... You could try to use a more modern version of MXNet, but then you'll need to adapt the code that examines the log file and scrapes all the number necessary for plotting from it. You could PR your changes that make the code compatible to a newer version if you like ;)
